lib/internal/fs/dir.js

```diff
@@ -132,6 +147,45 @@ class Dir {
     );
   }
 
+  processReadResult(path, result) {
+    for (let i = 0; i < result.length; i += 2) {
+      ArrayPrototypePush(
+        this[kDirBufferedEntries],
+        getDirent(
+          pathModule.join(path, result[i]),
```

Comment:
```suggestion
          path,
```
this fixes smartOs

Reply:
`(result[i + 1] === UV_DIRENT_UNKNOWN ? path : pathModule.join(path, result[i]))`